twingate/resource_group.go

```diff
@@ -96,11 +93,18 @@ func resourceGroupDelete(ctx context.Context, resourceData *schema.ResourceData,
 func resourceGroupRead(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*Client)
 
-	var diags diag.Diagnostics
-
 	groupID := resourceData.Id()
 	group, err := client.readGroup(ctx, groupID)
 
+	var groupName string
+	if group != nil {
+		groupName = string(group.Name)
+	}
+
+	return resourceGroupReadHelper(resourceData, groupID, groupName, err)
```

Comment:
This is incorrect - you always succeed.
What if `err` is `ErrGraphqlResultIsEmpty` ?  The resource doesn't exist on the server - you don't want to set the `id` in the state just with a null `name`